Make the download progress bar optional

evertrol opened the pull request:
This is a suggestion to make the download progress bar optional.

I've implemented this only for the IRSA dust extinction table utility for now. If this suggestion is well received, I'll try to implement it for other queries as well.

bsipocz commented:
Thanks @evertrol, this indeed something that makes sense to have. I wonder whether we should have ``cache`` as well?

Also the appveyor fail seems unrelated, I'll try to fix it in a separate PR.

evertrol commented:
I've updated the PR to include (hopefully all) the cases that use a progress bar.

A few notes:

- So far, I have implemented this by adding `show_progress` to all `FileContainer` uses, and then backwards through the various `get_..._async()` and `get_...()` methods.

- I have not updated the doc strings. Note that the `cache` keyword is (often) also not documented.

  - Side note: some `get_...()` methods now tend to have a multitude of keywords; perhaps too many? Maybe there's a use for `**kwargs` here instead (pop off the useful ones, pass the rest through to FileContainer)?

- I have not included any unit tests yet. Is there an easy way (perhaps through pytest) to test for output?
  The progress bar download comes in through astropy, so perhaps it's not astroquery to test this anyway.

- I've not touched `template_module/core.py`

- I haven't checked for other methods (such as the `get_images_list()` methods), since these don't appear to use `FileContainer`.

- Running the unit tests locally shows lots of errors when using Python 3.5, but there are no problems with 2.7. Is that correct?

keflavich commented:
I don't have answers for most of your questions, but I'm in favor of this change.  If I have time, I'll try to investigate the unit test errors, but don't expect that to happen soon.

bsipocz commented on astropy_helpers:
```diff
@@ -1 +1 @@
-Subproject commit 3e044eb26395c8569e3e15bf821c6ae5eeea324a
+Subproject commit fedd7c52ba80e2246f83b8d1752baa3a153a33e2
```
Could you please remove this change? I think it was accidental as the helpers are up to date in master.

evertrol replied:
Yes, that accidentally slipped through (it usually stays out of my way; looks like I needed to update it).

bsipocz commented:
@evertrol  - Do you see those errors for master, too? Travis passes for 3.5, so if you see any failures, could you open an issue with the logs?

evertrol commented:
My local unit test failure with 3.5 appears to be related to reading the VO tables: they fail with a recursion error in `astropy.utils.xml.iterparser.get_xml_iterator()`. I have no idea what's causing that, because writing the same input to file and then reading it outside running the unit test works fine. The `astropy.io.votable` tests themselves also pass.
Perhaps this sounds familiar to someone, since I'm running out of ideas how to fix it. For now, I'll trust the 2.7 tests and Travis.

keflavich commented:
That one is new to me, but I often have trouble with votable reading.  Sometimes it has turned out to be a bad encoding (bytes vs unicode), sometimes a bad specification in the votable, but if it's really the same input both times those are probably not the explanation...

evertrol commented:
The failing unit tests with 3.5 were the result of one of the more annoying and convoluted (dependency) bugs I've come across.
While I doubt anyone else will ever run into it, I'll note it here anyway, for the future (and possibly as a rant in disguise).

Firstly, since `_parse_result` catches any exception and turns it into a `TableParseError`, that needed to be turned off (harder to inspect the results/class instance in test mode). Unfortunately, the actual exception was a `RecursionError`, and `pytest` tries to be smart and shows only a limit number of lines in the backtrace; which needed to be increased to show the actual cause. Which was a recursion in `ndarray.__getitem__`, strangely enough.

Parsing a VO table does turn things into a (masked) `recarray`, but this `recarray` turned out to be a `pyfits.py3compat.recarray` (pyfits does some "smart" renaming, and it then shadows `numpy.recarray`). I had a (slightly) older version of pyfits, which was incompatible with recent numpy. Upgrading pyfits (or removing it entirely) made the error disappear.

Why was pyfits there? Because the `astroquery.alma` subpackage actually makes finding charts, using aplpy. And aplpy can use pyfits before astropy.io.fits (not sure why it does that, as it already relies on astropy).
And the astropy testrunner will inspect `astroquery.conftest`, and import the packages listed there (which includes `aplpy`, and then `pyfits` with its own `recarray` version). That also explains why I couldn't produce it as a standalone test: I didn't import `aplyp` or `pyfits`.

keflavich commented:
@evertrol that is an impressive trail.  Thanks for catching it.  Maybe raise an issue on aplpy to finally & permanently deprecate pyfits?

keflavich commented:
@evertrol Could you rebase this and confirm whether it's ready to merge?  A rebase should fix the appveyor failure.

evertrol commented:
Rebased; should be good to merge.
